### PR TITLE
[charts] Complete the docs to introduce charts

### PR DESCRIPTION
**Pull request by alexfauquette**
This PR is mostly about branding the package in the [overview](https://deploy-preview-9153--material-ui-x.netlify.app/x/react-charts/overview/)

### CODE REVIEW

File: docs/data/charts/overview/overview.md [hunk 1]
```diff
@@ -7,13 +7,30 @@ title: Charts - Overview
...
-💡 If you do not see the charts you want in the menu, we may have forget it.
-Please open an issue to ask for it, explaining which need it solves.
-:::
+## Overview
+
+The `@mui-x/chars` is an MIT library to render charts.
+It relies on D3 for data manipulation and SVG for rendering.
+
+Like other MUI components, charts are built to be production-ready components with nice integration into your application for common use cases.
+And provide a high level of customization.
```
Comment (alexfauquette) [inline docs/data/charts/overview/overview.md:18]:
```suggestion
They also provide a high level of customization.
```

**Review by flaviendelangle — APPROVED** [review]
Very nice improvements!
Alpha here we come!

File: docs/data/charts/styling/ColorTemplate.js [hunk 2]
```diff
@@ -5,22 +5,45 @@ import Stack from '@mui/material/Stack';
 import TextField from '@mui/material/TextField';
 import MenuItem from '@mui/material/MenuItem';
 import Typography from '@mui/material/Typography';
+import globalChance from 'chance';
+
+const chance = globalChance();
```
Comment (oliviertassinari) [inline docs/data/charts/styling/ColorTemplate.js:10]: This is flaky in Argos CI, e.g. https://app.argos-ci.com/mui/mui-x/builds/11605/49763612. I think that we could solve this like in the data grid:

https://github.com/mui/mui-x/blob/38733c626ea74819a6b0d90b7a2baa7ea8a024f1/packages/grid/x-data-grid-generator/src/services/random-generator.ts#L20-L22

**Comment by oliviertassinari** [comment]
> Alpha here we come!

@mbrookes left me feedback that it would be great to have a design pass on the chart demos.

**Comment by alexfauquette** [comment]
> Demos in the docs with a smaller height could be great, I could more easily navigate the docs. e.g. [recharts.org/en-US/api/AreaChart](https://recharts.org/en-US/api/AreaChart) is a small smaller in height

Effectively

> The tooltips feel a bit hard to read. I think that there is too much margin, and the font size is not always consistent or visible enough.

Effectively I did not fixed the same way tooltip for items and tooltips for axis

> Some of the colors could contrast more, e.g. [deploy-preview-9153--material-ui-x.netlify.app/x/react-charts/areas-demo/#stackedareachart](https://deploy-preview-9153--material-ui-x.netlify.app/x/react-charts/areas-demo/#stackedareachart)

Those colors are copy paster from recharts demo. But they will be replaced by a new palette in #9006

**Comment by oliviertassinari** [comment]
More ideas:

2.a. On https://deploy-preview-9006--material-ui-x.netlify.app/x/react-charts/scatter/#basics it's a bit hard to get the tooltip shown for the data points, it's much simpler on https://observablehq.com/plot/marks/tip#tip-options (testing the points at the edge). I think that we could have some sort of a proximity trigger.
2.b. https://www.tremor.so/docs/components/bar-chart design makes sense to me, e.g.

Screenshot 2023-06-15 at 22 54 58

2.c. To consider if we don't want different tooltip modes, one that follows the cursor, and another that follows the data. What they have in https://observablehq.com/plot/marks/tip#tip-options is quite advanced.

4. Something we could look in the future is animations.
5. We have a rechart chart in https://mui.com/material-ui/customization/typography/#responsive-font-sizes, something we could dogfood with for our library at one point.
6. The legend is cut in https://deploy-preview-9153--material-ui-x.netlify.app/x/react-charts/lines/#stacking

Screenshot 2023-06-15 at 23 12 47

**Comment by alexfauquette** [comment]
@oliviertassinari All the feedbacks of your first message are here, plus a few others I spotted while doing it (your point 6 "axis's label is cut" is one of them)

https://github.com/mui/mui-x/pull/9341

For the other comments, I will open dedicated issues to track them

**Comment by flaviendelangle** [comment]
> 5 Dog fooding. We have a rechart chart in https://mui.com/material-ui/customization/typography/#responsive-font-sizes, something we could replace with our charts at one point.

That could be a nice little test to let someone from the core do it and ask him/her the pain points he/she had.

**Pull request merged**